test/Templates.Test/Helpers/AspNetProcess.cs

```diff
@@ -48,9 +61,12 @@ public AspNetProcess(ITestOutputHelper output, string workingDirectory, string p
                     .WaitForExit(assertSuccess: true);
             }
 
+            _httpPort = 5000 + rnd.Next(2000);
+            _httpsPort = 44300 + rnd.Next(1000);
             var envVars = new Dictionary<string, string>
             {
-                { "ASPNETCORE_URLS", "http://127.0.0.1:0" }
+                { "ASPNETCORE_URLS", $"http://localhost:{_httpPort};https://localhost:{_httpsPort}" },
```

Comment:
I had to switch the strategy for picking up the ports due to the fact that Kestrel won't allow me to let the OS pick the port when binding to localhost and that we need to bind to localhost for HTTPS requests to work.

Reply:
I'll switch this back to 127.0.0.1:0 once we have aspnet/BasicMiddleware#275 and alternatively we will set the _httpClient.BaseAddress to http://localhost:<<selected-port>>

Reply:
I know there's a relatively low probability of port collisions with this logic, but it wouldn't be much harder to ensure noncollision just by using an incrementing static variable to make it choose ports 5000, 5001, 5002, etc.

Reply:
Yep, i've already seen this approach cause issues, so I'm going to switch to pick a "random base port" and then pick incremental ports with a static variable